### DISCUSSION TITLE
Remix implementation

### DISCUSSION THREAD

**Discussion by juan-manuel-alberro** [mantinedev/mantine#600] [Q&A] [answered]
Hello guys,

Has anybody tried this fav library with [Remix](https://remix.run/)? This SSR framework comes open-source a couple of weeks ago and it's a huge competitor for Next.js (and better in my opinion).

Thanks!

**Comment by rtivital** [comment, marked as answer]
I've tried adding support but there were some issues – https://github.com/rtivital/remix-mantine-ssr
You can find information about general ssr approach here – https://mantine.dev/theming/ssr/

**Reply by juan-manuel-alberro** [reply]
Thanks Vitaly, I'll take a look and share some insights as well.

**Reply by jaaneh** [reply]
Any further insights on this? Would love to get this working as well.

**Reply by rtivital** [reply]
See https://github.com/rtivital/remix-mantine-ssr, it works fine

**Reply by jaaneh** [reply]
> See https://github.com/rtivital/remix-mantine-ssr, it works fine

Appreciate you, thanks! 🎉

**Reply by juan-manuel-alberro** [reply]
Hi guys, I'm having an issue when I use `Link` from Remix inside a `Group`

```jsx
        <Group position="apart" mt="xl">
          Don't have an account?{" "}
          <Link
            to="/auth/sign-up"
            className="text-info text-gradient font-weight-bold"
          >
            Sign up
          </Link>
          <Button type="submit">Login</Button>
        </Group>
```

And the error is 

```
TypeError: Cannot read property 'className' of undefined
    at /node_modules/@mantine/core/cjs/components/Group/Group.js:77:46
    at Array.map (<anonymous>)
    at Object.@mantine/core/Group [as render] (/node_modules/@mantine/core/cjs/components/Group/Group.js:76:34)
    at ReactDOMServerRenderer.render (/node_modules/react-dom/cjs/react-dom-server.node.development.js:3872:44)
    at ReactDOMServerRenderer.read (/node_modules/react-dom/cjs/react-dom-server.node.development.js:3690:29)
    at renderToString (/node_modules/react-dom/cjs/react-dom-server.node.development.js:4298:27)
    at handleRequest (/build/index.js:419:51)
    at renderDocumentRequest (/node_modules/@remix-run/server-runtime/server.js:404:18)
    at async requestHandler (/node_modules/@remix-run/server-runtime/server.js:55:20)
    at async /node_modules/@remix-run/express/server.js:45:22
```

Is something reproducible for you as well?

**Reply by rtivital** [reply]
You cannot render strings inside group, wrap it with span or div: `Don't have an account?{" "}`

**Reply by juan-manuel-alberro** [reply]
Thanks @rtivital that was the issue, I wasn't aware of that limitation in the `Group`, you rock!

**Comment by larissa-n** [comment]
If anyone is hosting Remix on Cloudflare Workers (so node features unavailable), a decent write-up of how to make that work: https://medium.com/@denzity/how-to-make-remix-work-with-mantine-cloudflare-pages-workers-907888bc00fc

**Reply by Viranchee** [reply]
This was helpful thank you!
